asserts/snapasserts/validation_sets.go

```diff
@@ -1031,33 +990,146 @@ func (v *ValidationSets) SnapConstrained(snapRef naming.SnapRef) bool {
 	return v.constraintsForSnap(snapRef) != nil
 }
 
-// CheckPresenceInvalid returns the list of all validation sets that declare
-// presence of the given snap as invalid. PresenceConstraintError is returned if
-// presence of the snap is "optional" or "required".
-// The method assumes that validation sets are not in conflict.
-func (v *ValidationSets) CheckPresenceInvalid(snapRef naming.SnapRef) ([]ValidationSetKey, error) {
-	cstrs := v.constraintsForSnap(snapRef)
+// ContainerPresence represents the allowed presence of a snap or component with
+// respect to a set of validation sets that it was derived from.
+type ContainerPresence struct {
```

Comment:
we have a PresenceConstraintError alread,y maybe this should be called PresenceContraint ?